app/connectors/SoftDrinksIndustryLevyConnector.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import play.api.Configuration
+import uk.gov.hmrc.http.HttpReads.Implicits.{readFromJson, _}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import models.retrieved.RetrievedSubscription
+
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SoftDrinksIndustryLevyConnector @Inject()(
+    val http: HttpClient,
+    val configuration: Configuration
+  )(implicit ec: ExecutionContext)
+  extends ServicesConfig(configuration) {
+
+  lazy val sdilUrl: String = baseUrl("soft-drinks-industry-levy")
+
+  private def getSubscriptionUrl( sdilNumber: String,identifierType: String = "sdil"): String = s"$sdilUrl/subscription/$identifierType/$sdilNumber"
+
+  def retrieveSubscription(
+    sdilNumber: String,
+    identifierType: String
+  )(implicit hc: HeaderCarrier): Future[Option[RetrievedSubscription]] =
+  http.GET[Option[RetrievedSubscription]](getSubscriptionUrl(sdilNumber: String,identifierType)).flatMap {
+    case Some(a) => Future(Some(a))
+    case _ => Future.successful(None)
+  }
```

Comment:
NABD , we can use map instead of flatMap i think
`http.GET[Option[RetrievedSubscription]](getSubscriptionUrl(sdilNumber: String,identifierType)).map {
    case Some(a) => identity
    case _ => None
  }`